### PR TITLE
add rules for new cases OCP-26823,OCP-26633,OCP-28661,OCP-23707,OCP-24004,OCP-24006,OCP-24002,OCP-26640

### PR DESCRIPTION
**Pull request by yuwang-RH**
add rules for new cases OCP-26823,OCP-26633,OCP-28661,OCP-23707,OCP-24004,OCP-24006,OCP-24002,OCP-26640

### CODE REVIEW

**Comment by yuwang-RH** [comment]
@yasun1 @xueli181114 @tzhou5  Please help review this PR, Thanks.

File: lib/rules/web/ocm_console/cluster_idp.xyaml [hunk 1]
```diff
@@ -0,0 +1,680 @@
+# --- the elements and actions about the idp configuration
+check_idp_hint:
+  elements:
+  - selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__icon']
+    timeout: 100
```
Comment (yasun1) [inline lib/rules/web/ocm_console/cluster_idp.xyaml:6]: is not the timeout too long?

File: lib/rules/web/ocm_console/cluster_idp.xyaml [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+  - selector:
+      xpath: //div[@id='idpHint']/h4[@class='pf-c-alert__title' and contains(text(),'Missing Identity Providers')]
+  - selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__description' and contains(text(),'Identity Providers determine how users log into the cluster')]
+  action: check_add_oauth_configuration_button
+check_add_oauth_configuration_button:
+  element: &oauth_configuration_button
+    selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__description']/button[text()='Add OAuth Configuration']
```
Comment (yasun1) [inline lib/rules/web/ocm_console/cluster_idp.xyaml:15]: Simple the rule to `//div[@id='idpHint']//button[text()='Add OAuth Configuration']`

**Review by yasun1 — COMMENTED** [review]
/lgtm

**Comment by yasun1** [comment]
Please update your name of the MR

File: lib/rules/web/ocm_console/cluster_idp.xyaml (hunk 2 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/cluster_idp.xyaml:15]: @yasun1  The part of "div[@class='pf-c-alert__description']"  indicates the button is under the alert message field, I think it is not a redundancy xpath part and we should keep it.

File: lib/rules/web/ocm_console/cluster_idp.xyaml (hunk 1 above)
Comment (yuwang-RH) [reply lib/rules/web/ocm_console/cluster_idp.xyaml:6]: @yasun1  I met failure when I use timeout 50, then succeed when I set timeout 100. It wouldn't take too long time actually~

**Pull request merged**